pkg/storegateway/series_chunks_test.go

```diff
@@ -722,7 +799,11 @@ func TestLoadingSeriesChunksSetIterator(t *testing.T) {
 			if testCase.expectedErr != "" {
 				assert.Zero(t, chunkBytesSlicePool.(*pool.TrackedPool).Gets.Load())
 			} else {
-				assert.Greater(t, chunkBytesSlicePool.(*pool.TrackedPool).Gets.Load(), int64(0))
+				if testCase.expectNoChunkBytesSlicePoolLoads {
```

Comment:
Sorry to only now get to this - do i understand correctly that `expectNoChunkBytesSlicePoolLoads` should be set when the result set is actually empty and there were no chunks to load? 

WDYT about doing the check if `len(expectedSets) == 0`, then don't expect allocated bytes from the pool?

Reply:
Good idea!